src/ui/components/results/directanswercomponent.js

```diff
@@ -237,6 +221,150 @@ export default class DirectAnswerComponent extends Component {
+      return this._getCardBasedOnOverrides({
+        directAnswer: directAnswer,
+        overrides: this._cardOverrides,
+        fallback: this._defaultCard
+      });
+    } else {
+      return this._defaultCard;
+    }
+  }
+
+  /**
+   * Determines the card that should be used based on the types option
+   *
+   * @param {Object} directAnswer The direct answer state
+   * @returns {string}
+   */
+  _getCardBasedOnTypes (directAnswer) {
+    if (!('type' in directAnswer) || !(directAnswer.type in this._types)) {
+      return this._defaultCard;
+    }
+
+    const typeOptions = this._types[directAnswer.type];
+
+    const cardFallback = typeOptions.cardType || this._defaultCard;
+
+    if (typeOptions.cardOverrides) {
+      return this._getCardBasedOnOverrides({
+        directAnswer: directAnswer,
+        overrides: typeOptions.cardOverrides,
+        fallback: cardFallback
+      });
+    }
+
+    return cardFallback;
+  }
+
+  /**
+   * Returns the custom card type that should be used for the given direct answer.
+   *
+   * @param {Object} directAnswer The direct answer state
+   * @param {Object[]} overrides The overrides to search through
+   * @param {string} fallback The card to return if no match is found
+   * @returns {string}
+   */
+  _getCardBasedOnOverrides ({ directAnswer, overrides, fallback }) {
+    const cardOverride = overrides.find(override => {
+      return this._overrideMatchesAnswer(override, directAnswer);
+    });
+    return cardOverride ? cardOverride.cardType : fallback;
+  }
+
+  /**
+   * Check whether a given cardOverride matches a given directAnswer.
+   *
+   * @param {Object} override
+   * @param {Object} directAnswer
+   * @returns {boolean}
+   */
+  _overrideMatchesAnswer (override, directAnswer) {
+    if (!Object.keys(directAnswer).length) {
+      return true;
+    }
+    const directAnswerPropeties = {
+      type: directAnswer.type,
+      entityType: directAnswer.relatedItem.data.type,
+      fieldName: directAnswer.answer.fieldName,
+      fieldType: directAnswer.answer.fieldType
+    };
+    for (let [propertyToMatch, propertyValue] of Object.entries(override)) {
+      if (propertyToMatch === 'cardType') {
+        continue;
+      }
+      if (directAnswerPropeties[propertyToMatch] !== propertyValue) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  /**
+   * Throws an error if the types config option is not formatted properly.
+   * @throws if validation fails
+   */
+  _validateTypes () {
+    if (!this._types) {
+      return;
+    }
+
+    Object.entries(this._types).forEach(([directAnswerType, typeOptions]) => {
+      this._validateTypeOptions(typeOptions);
+      typeOptions.cardOverrides && this._validateCardOverrides(typeOptions.cardOverrides);
+    });
+  }
+
+  /**
+   * Throws an error if any typeOptions includes unsupported keys
+   *
+   * @param {Objet} typeOptions
+   * @throws if validation fails
+   */
+  _validateTypeOptions (typeOptions) {
```

Comment:
nit: would it be a little more concise to include these 3 methods into _validateTypes() instead? _validateSupportedKeysOfObject could be an inline function if you don't want to repeat things

Reply:
sure